docs/content/docs/concepts/Breaking Changes and Supported Patterns.md

```diff
@@ -0,0 +1,293 @@
+  Can be a compiler error, runtime error, or causing previously Supported User Code to become unsupported.
+  For example a Change could break user code by making it not compile, crash, no longer be Supported User Code, return incorrect results etc.
+- Backwards Compatible Change: A change to The Package which cannot not Break any possible Supported User Code.
+- Incompatible Change: A Change to The Package which could break a Supported User.
+
+The goal of this document is to justify having "Supported Patterns" as defined above, and to suggest some such patterns and explain their motivations.
+
+## Why have Supported Patterns?
+
+To help the developers of the package communicate clearly about how their changes impact users, they need to make some assumptions.
+These assumptions are the Supported Patterns.
+
+One of the most important details the developers of the package should communicate is which changes are Backwards Compatible and which ones are Incompatible.
+A common convention in this space is to use [Sematic Versioning](https://semver.org/) to communicate this via the version of the package, but such communication is important even if not using Sematic Versioning.
+
+For this communication to be meaningful both sides have to agree on terminology.
+The challenging part of this is providing a precise definition of what is a Backwards Compatible Change and what is an Incompatible Change.
+Supported Patterns provide this definition.
+
+Additionally, if the supported patterns are co-designed with recommended design patterns and policies for developing the package, it can help developers of the package reduce the frequency of Incompatible Changes.
+
+The challenge is to pick Supported Patterns and associated design patterns for the package that:
+
+1. Maximize the ease of making user code supported, and make it easy to tell if user code is supported.
+2. Make it easy for developers of the package to know if a change is Incompatible.
+3. Maximize package developer's ability to make useful changes be Backwards Compatible.
+4. Maximize ease of adopting Incompatible Changes in Supported User Code. For example use the compiler to inform Supported Users where they have to update their code to reduce the risk of broken runtime behavior if the necessary adjustment is missed.
+
+## Why Trivial Supported Patterns are not enough:
+
+If deciding all patterns are supported then all changes are Incompatible Changes.
+A trivial proof of this is a Supported User could compute a checksum of the package to validate it was loaded correctly.
+Such a User would be broken by any possible package change.
+
+Another simple approach would be to define Supported Patterns as TypeScript code which uses the package APIs in such a way that it compiles.
+Unfortunately doesn't solve the problem: nearly everything is still incompatible change since TypeScript lets you traverse modules objects, inspect the code implementing functions etc.
```

Comment:
```suggestion
Unfortunately, this doesn't solve the problem: nearly everything is still incompatible change since TypeScript lets you traverse modules objects, inspect the code implementing functions etc.
```